delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/QueryGraph.scala

```diff
@@ -8,22 +8,57 @@ import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
 import ch.epfl.bluebrain.nexus.delta.rdf.graph.{Graph, NTriples}
 import ch.epfl.bluebrain.nexus.delta.rdf.query.SparqlQuery.SparqlConstructQuery
 import ch.epfl.bluebrain.nexus.delta.sourcing.state.GraphResource
+import fs2.Chunk
 import monix.bio.Task
 
 import java.util.regex.Pattern.quote
 
 /**
-  * Pipe that performs the provided query for the incoming resource and replaces the graph with the result of query
+  * Provides a way to query for the multiple incoming resources (from a chunk). This assumes that the query contains the
+  * template: `VALUE ?id { {resource_id} }`. The result is a single Graph for all given resources.
   * @param client
-  *   the blazegraph client
+  *   the blazegraph client used to query
+  * @param namespace
+  *   the namespace to query
+  * @param query
+  *   the sparql query to perform
+  */
+final class BatchQueryGraph(client: BlazegraphClient, namespace: String, query: SparqlConstructQuery) {
```

Comment:
Now, they can be split in their own files if the trait is gone :)

Reply:
Done 🔪 😅